InnerEye/ML/Histopathology/models/deepmil.py

```diff
@@ -187,7 +191,7 @@ def log_metrics(self,
                 log_on_epoch(self, f'{stage}/{metric_name}', metric_object)
 
     def forward(self, images: Tensor) -> Tuple[Tensor, Tensor]:  # type: ignore
-        with no_grad():
+        with set_grad_enabled(self.is_finetune):
             H = self.encoder(images)                        # N X L x 1 x 1
```

Comment:
One additional thought: here we should call the function to encode in chunks, otherwise we will be limited in the max_bag_size as we were back at the beginning of our experiments

Reply:
I think doing this is more involved as monai transforms use dataset and dataloaders and it could be a separate PR.